php_phongo.c

```diff
@@ -1377,10 +1377,7 @@ void php_phongo_write_concern_to_zval(zval* retval, const mongoc_write_concern_t
 	}
 
 	if (wtimeout != 0) {
-		/* Note: PHP currently enforces that wimeoutMS is a 32-bit integer, so
-		 * casting will never truncate the value. This may change with
-		 * PHPC-1411. */
-		ADD_ASSOC_LONG_EX(retval, "wtimeout", (int32_t) wtimeout);
+		ADD_ASSOC_LONG_EX(retval, "wtimeout", wtimeout);
```

Comment:
`php_phongo_write_concern_to_zval` only appears to be called from `php_phongo_bulkwrite_get_debug_info`. Given that, I think it'd make sense to use the same logic here as we do in [`php_phongo_cursorid_get_debug_info`](https://github.com/mongodb/mongo-php-driver/blob/1.6.0/src/MongoDB/CursorId.c#L100) and conditionally output `wtimeout` as a string if we're on a 32-bit architecture _and_ its value exceeds `INT32_MAX` (admittedly an edge case). `ADD_ASSOC_LONG_EX` works for all other cases.

Was your original thinking that a user debugging a WriteConcern object would have created it on the same machine, and in that case the value itself would naturally be within the range of 32-bit positive integers?

This may relate to my later comment on `PHP_METHOD(WriteConcern, getWtimeout)`.

Reply:
Updated this to reflect the logic in serialisation of CursorId and Int64: this will be exposed as string on 32-bit systems.

Reply:
My apologies. CursorId does this in `php_phongo_cursorid_get_debug_info`. Changing this to also return a LONG as long as it's within 32-bit boundaries works for me 👍